# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dovecot/dovecot-2.2.1-0001-upstream-46dc61f6acae.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---


# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1366370963 -10800
# Node ID 46dc61f6acae8b441ec2b131fa10d6747f194dde
# Parent  999564a5b2a57e13edb04b6cb211349c8d9fc30d
Compiling fix for Sun compilers.
I wish gcc/clang warned about these as well, as sometimes they indicate bugs.

diff -r 999564a5b2a5 -r 46dc61f6acae src/imap-urlauth/imap-urlauth-worker.c
--- a/src/imap-urlauth/imap-urlauth-worker.c	Fri Apr 19 00:38:05 2013 +0300
+++ b/src/imap-urlauth/imap-urlauth-worker.c	Fri Apr 19 14:29:23 2013 +0300
@@ -941,7 +941,7 @@
 {
 	bool debug;
 
-	debug = getenv("DEBUG");
+	debug = getenv("DEBUG") != NULL;
 	access_user = access_user != NULL ? access_user : getenv("USER");
 	if (access_user == NULL && IS_STANDALONE())
 		access_user = getlogin();
diff -r 999564a5b2a5 -r 46dc61f6acae src/lib/ioloop.c
--- a/src/lib/ioloop.c	Fri Apr 19 00:38:05 2013 +0300
+++ b/src/lib/ioloop.c	Fri Apr 19 14:29:23 2013 +0300
@@ -663,7 +663,7 @@
 
 bool io_loop_have_ios(struct ioloop *ioloop)
 {
-	return ioloop->io_files;
+	return ioloop->io_files != NULL;
 }
 
 bool io_loop_have_immediate_timeouts(struct ioloop *ioloop)

